src/main/java/com/universeprojects/cacheddatastore/CachedDatastoreService.java

```diff
@@ -686,7 +686,23 @@ public CachedEntity getIfExists(Key entityKey)
 			return null;
 		}
 	}
-	
+
+	public CachedEntity refetch(Key entityKey)
+	{
+		if (entityKey==null)
+			throw new IllegalArgumentException("Key cannot be null.");
+		if (entityKey.isComplete()==false)
+			throw new IllegalArgumentException("The entity you are attempting to refetch hasn't even been saved to the DB yet as the key is incomplete.");
+		try
+		{
+			return get(entityKey);
+		}
+		catch(EntityNotFoundException ise)
+		{
+			throw new IllegalStateException("Entity "+entityKey+" was not found in the database.", ise);
+		}
+	}
+
 	public CachedEntity refetch(CachedEntity entityToRefetchFromDB) 
```

Comment:
the old method should actually call the new refetch(Key) method instead to avoid code duplication